### PR TITLE
Use new `createContext`, remove `getChildContext`. Add `useText` hook.

### PR DESCRIPTION
**Pull request by pl12133**
Resolves https://github.com/synacor/preact-i18n/issues/41 and https://github.com/synacor/preact-i18n/issues/39.

This continues the upgrade to Preact X, moving away from legacy context into new context.

This will maintain the same interface while converting from legacy context to `createContext`. However this will be a breaking change in the case of directly accessing `this.context.intl` which will now be `undefined`.

### CODE REVIEW

**Comment by IrregularShed** [comment]
@pl12133 Hi Peter, have you got any further with this? I'd like to help in any way I can to get this over the line.

**Comment by jliebrand** [comment]
+1 would love to see preact-i18n working with preactX

**Comment by pl12133** [comment]
I'm glad to hear this PR is in demand! I'm going to take a look at the test cases today and once they pass this should be ready to merge and release as a beta.

It would be super helpful if people could check out this branch and try it in their own projects, and also try out the new `useText` hook and see if any bugs come up.

**Comment by pl12133** [comment]
Old and new tests are passing, this should be ready for review.

If anyone could try this out in their own applications and let us know about any issues that would be very helpful.

**Comment by IrregularShed** [comment]
@pl12133 As soon as I get a chance to, I'll give it a try. At the moment I've found myself having to use i18next to keep the powers that be happy but I'll branch off an earlier commit and see how it goes.

File: README.md [hunk 1]
```diff
@@ -374,6 +375,43 @@ const WrappedFoo = withText('user.placeholer')(Foo);
...
+	const { placeholder } = useText({
+		placeholder: 'user.placeholder'
+	});
+	return <input placeholder={placeholder} />
+}
+```
+
+```javascript
+function MyComponent() {
+	const { placeholder } = withText({
```
Comment (desouzas) [inline README.md:399]:
```suggestion
	const { placeholder } = useText({
```

File: src/hooks/use-text.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+ *
+ *	@example function MyComponent() {
+ *		const { placeholder } = useText({
+ *			placeholder: 'user.placeholder'
+ *		});
+ *		return <input placeholder={placeholder} />
+ *	}
+ *
+ *	@example function MyComponent() {
+ *		const { placeholder } = withText({
```
Comment (desouzas) [inline src/hooks/use-text.js:17]:
```suggestion
 *		const { placeholder } = useText({
```

**Pull request merged**